Zilon.Core/Zilon.Core/Tactics/Behaviour/HumanActorTaskSource.cs

```diff
@@ -1,57 +1,165 @@
 ﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Zilon.Core.Tactics.Behaviour
 {
     public class HumanActorTaskSource : IHumanActorTaskSource
     {
-        private IActorTask _currentTask;
-        private IIntention _currentIntention;
+        private SpscChannel<IActorTask> _spscChannel;
 
-        public void SwitchActor(IActor currentActor)
+        public HumanActorTaskSource()
         {
-            CurrentActor = currentActor;
+            _spscChannel = new SpscChannel<IActorTask>();
         }
 
-        public IActor CurrentActor { get; private set; }
-
+        public HumanActorTaskSource(IActor activeActor) : this()
+        {
+            SwitchActiveActor(activeActor);
+        }
 
-        public void Intent(IIntention intention)
+        public void SwitchActiveActor(IActor currentActor)
         {
-            _currentIntention = intention ?? throw new ArgumentNullException(nameof(intention));
+            ActiveActor = currentActor;
         }
 
-        public IActorTask[] GetActorTasks(IActor actor)
+        public IActor ActiveActor { get; private set; }
+
+        private bool _intentionWait;
+
+        public Task IntentAsync(IIntention intention)
         {
-            if (actor != CurrentActor)
+            if (_intentionWait)
             {
-                return Array.Empty<IActorTask>();
+                throw new Exception();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5544638)